cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -113,8 +113,8 @@ public void configure(Map<String, ?> configs) {
 
     //Add BootstrapServers if not set
     if (!producerProps.containsKey(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG)) {
-      Object port = configs.get("port");
-      String bootstrapServers = "localhost:" + (port == null ? "9092" : String.valueOf(port));
+      String listeners = configs.get("listeners") == null ? null : String.valueOf(configs.get("listeners"));
+      String bootstrapServers = listeners == null ? "localhost:9092" : listeners.substring(listeners.lastIndexOf("/") + 1, listeners.length());
```

Comment:
To ensure backwards compatibility and robustness, can we ensure that
1.  if `configs.get("port")` is not `null`, use this as port
2. otherwise if `configs.get("listeners")` is not `null` and it has a non-empty `port`, use that as port
3. otherwise use the default hostname and port -- i.e. `localhost` and `9092`.

Also can we move the parsing of port to a small function with this priority of retrieval documented?